Transition development to Orbiter 2024 x64, change build system to CMake

CaptainSwag101 opened the pull request:
I believe that NASSP would gain significant benefits from transitioning our build system to use CMake. In a nutshell, CMake is a higher-level, more abstract system than something like the Visual Studio project files that we currently use to define our vessels, libraries, MFDs, etc. Rather than directly instructing a C++ compiler on how to build the code, it processes a set of instructions like a script, and generates an intermediary output file which is then used *by* a C++ compiler to build our code. (In theory, CMake can actually be used to *generate* VS project files, which is pretty cool.)

CMake has some very nice features for those familiar with it, but it will also let us keep our existing workflow relatively unchanged. Some particular benefits I see for NASSP are as follows:

- Having much less verbose, human-readable project files that can be edited like source code, and which will play nicer with git.
- The ability to use logic within our project files and introduce/remove in-code definitions directly from the build system, depending on user-defined settings or for particular build configurations. As an example, while we would eventually like to not require explicit code changes to enable J-Mission systems, CMake could (with some work) allow toggling J-Mission code and systems configs with a single check-box, while we figure out a more elegant and modular solution down the line.
- It gives our project files a "clean slate" free of the roughly 20 years of legacy leftovers and inscrutable per-project tweaks and adjustments. This also helps make them easier to understand as a side-effect.
- More progress towards an OS-agnostic future. While CMake is compatible with Visual Studio, and can even be used to generate VS project files if desired, moving away from direct reliance on Visual Studio will give us more flexibility going forward with what tools we use, and if Orbiter ever makes a full transition away from Windows-specific SDKs and libraries, this will make it easier for us to follow suit.
- Significantly less work is required to enable OpenOrbiter/Orbiter 2024 compatibility, with the added benefit of being able to keep VESIM functionality in all situations. OpenOrbiter/Orbiter Beta detection is now done automatically based on the presence of "lua.dll", which should only exist in OpenOrbiter/Orbiter 2024 installations because it's named differently in Orbiter Beta. No more separate branches and frustrating merge conflicts; now one branch can build for both versions of the simulator.

Since these changes would inherently allow for OpenOrbiter compatibility, this PR supersedes #793.

I have had several devs personally encourage these changes, and I am naturally an advocate for switching to CMake after seeing the workflow and practical improvements it offers us. As-is, the only changes needed to put this in a fully "ready for prime-time" state would be to delete the old VS project files, to ensure users don't get confused and use the wrong system. I have not done this yet, for the sake of making it easy to merge/rebase to upstream changes or for folks to experiment with migrating their outstanding pull requests if they so desire. These are foundational changes, and I want to give everybody plenty of time to figure out what this is all about, and to test and give me feedback on workflow, stability, and other issues that I can't discover by myself. I have only been using and learning about CMake for a year or two, and this is by far the most complex thing I've done with it. Nevertheless, I have been very encouraged by how simple it has been to re-organize a project as big and complex as NASSP into this new build system with relatively little work; I think it speaks to CMake's strengths that it can be done by even someone as relatively inexperienced as me.

CaptainSwag101 committed:
Restart OO development from scratch with CMake
I'm aiming for seamless cross-compatibility between Orbiter Beta R90 and OpenOrbiter, let's see if it's possible!

CaptainSwag101 committed:
Restart OO development from scratch with CMake
I'm aiming for seamless cross-compatibility between Orbiter Beta R90 and OpenOrbiter, let's see if it's possible!

CaptainSwag101 committed:
Restart OO development from scratch with CMake
I'm aiming for seamless cross-compatibility between Orbiter Beta R90 and OpenOrbiter, let's see if it's possible!

CaptainSwag101 committed:
Add remaining projects
Everything is compiling successfully!

CaptainSwag101 committed:
Add remaining projects
Everything is compiling successfully!

CaptainSwag101 committed:
Add remaining projects
Everything is compiling successfully!

CaptainSwag101 committed:
Fix MCC vessel build output location

CaptainSwag101 committed:
Fix MCC vessel build output location

CaptainSwag101 committed:
Fix MCC vessel build output location

CaptainSwag101 committed:
Add compatibility with Orbiter Beta R90

CaptainSwag101 committed:
Add compatibility with Orbiter Beta R90

CaptainSwag101 committed:
Add compatibility with Orbiter Beta R90

CaptainSwag101 committed:
Delete CMakeSettings.json

CaptainSwag101 committed:
Delete CMakeSettings.json

CaptainSwag101 committed:
Delete CMakeSettings.json

CaptainSwag101 committed:
Add new default CMakeSettings.json
This should hopefully be all sane defaults, which we won't change unless we have to.

CaptainSwag101 committed:
Add new default CMakeSettings.json
This should hopefully be all sane defaults, which we won't change unless we have to.

CaptainSwag101 committed:
Add new default CMakeSettings.json
This should hopefully be all sane defaults, which we won't change unless we have to.

CaptainSwag101 committed:
Ignore further changes to CMakeSettings.json
This should avoid myself (or other people) from accidentally committing situational config options based on their personal installations of NASSP/Orbiter.

CaptainSwag101 committed:
Ignore further changes to CMakeSettings.json
This should avoid myself (or other people) from accidentally committing situational config options based on their personal installations of NASSP/Orbiter.

CaptainSwag101 committed:
Ignore further changes to CMakeSettings.json
This should avoid myself (or other people) from accidentally committing situational config options based on their personal installations of NASSP/Orbiter.

CaptainSwag101 committed:
More gitignore changes

CaptainSwag101 committed:
More gitignore changes

CaptainSwag101 committed:
More gitignore changes

CaptainSwag101 committed:
Un-hardcode project name into linking details

CaptainSwag101 committed:
Un-hardcode project name into linking details

CaptainSwag101 committed:
Un-hardcode project name into linking details

CaptainSwag101 committed:
Add new Ascent/Rendezvous Monitor source files to projects

CaptainSwag101 committed:
Add new Ascent/Rendezvous Monitor source files to projects

CaptainSwag101 committed:
Add new Ascent/Rendezvous Monitor source files to projects

CaptainSwag101 committed:
Add new setting for final output directory for special installations

CaptainSwag101 committed:
Add new setting for final output directory for special installations

CaptainSwag101 committed:
Add new setting for final output directory for special installations

CaptainSwag101 committed:
Configure config/data file auto-copying

CaptainSwag101 committed:
Configure config/data file auto-copying

CaptainSwag101 committed:
Configure config/data file auto-copying

CaptainSwag101 committed:
Add extra check for final dir == repo root before copying

CaptainSwag101 committed:
Add extra check for final dir == repo root before copying

CaptainSwag101 committed:
Add extra check for final dir == repo root before copying

CaptainSwag101 committed:
Clarify copy variable name, fix NASSP-inside-Orbiter-dir detection

CaptainSwag101 committed:
Clarify copy variable name, fix NASSP-inside-Orbiter-dir detection

CaptainSwag101 committed:
Clarify copy variable name, fix NASSP-inside-Orbiter-dir detection

CaptainSwag101 committed:
Reorganize main CMakeLists file

CaptainSwag101 committed:
Reorganize main CMakeLists file

CaptainSwag101 committed:
Reorganize main CMakeLists file

CaptainSwag101 committed:
Message the final install dir when copying

CaptainSwag101 committed:
Message the final install dir when copying

CaptainSwag101 committed:
Message the final install dir when copying

thewonderidiot commented on Orbitersdk/samples/ProjectApollo/Project/LEM/CMakeLists.txt:
```diff
@@ -0,0 +1,109 @@
+target_link_libraries(${PROJECT_NAME} Orbiter)
+target_link_libraries(${PROJECT_NAME} Orbitersdk)
+target_link_libraries(${PROJECT_NAME} PanelSDK)
+target_link_libraries(${PROJECT_NAME} XRSound)
```
Is there a reason why you're only passing one library to each `target_link_libraries()` call? You could instead do something like
```
target_link_libraries(${PROJECT_NAME} PRIVATE
    GLU32
    dxguid
    dinput8
    OpenGL32
    User32
    WinMM
    WS2_32
    Orbiter
    Orbitersdk
    PanelSDK
    XRSound
)
```
which would be less verbose and potentionally a bit more performant on the CMake side.

CaptainSwag101 replied:
Ahh, I didn't realize that was possible, thanks! Much more elegant.

CaptainSwag101 committed:
Consolidate target_link_libraries() statements in projects

CaptainSwag101 committed:
Consolidate target_link_libraries() statements in projects

CaptainSwag101 committed:
Consolidate target_link_libraries() statements in projects

CaptainSwag101 committed:
Vastly simplify install step, only copy data files when explicitly running "Install"

CaptainSwag101 committed:
Vastly simplify install step, only copy data files when explicitly running "Install"

CaptainSwag101 committed:
Vastly simplify install step, only copy data files when explicitly running "Install"

CaptainSwag101 committed:
Enable auto-detection of OpenOrbiter, optimize install step slightly

CaptainSwag101 committed:
Enable auto-detection of OpenOrbiter, optimize install step slightly

CaptainSwag101 committed:
Enable auto-detection of OpenOrbiter, optimize install step slightly

CaptainSwag101 committed:
Move MSVC compiler settings to after OpenOrbiter flag definition

CaptainSwag101 committed:
Move MSVC compiler settings to after OpenOrbiter flag definition

CaptainSwag101 committed:
Move MSVC compiler settings to after OpenOrbiter flag definition

CaptainSwag101 committed:
Fix arrangement of Orbiter install dir and final install dir variables

CaptainSwag101 committed:
Fix arrangement of Orbiter install dir and final install dir variables

CaptainSwag101 committed:
Fix arrangement of Orbiter install dir and final install dir variables

CaptainSwag101 committed:
Actually fix variable ordering this time

CaptainSwag101 committed:
Actually fix variable ordering this time

CaptainSwag101 committed:
Actually fix variable ordering this time

CaptainSwag101 committed:
Copy Missions folder during Install step

CaptainSwag101 committed:
Copy Missions folder during Install step

CaptainSwag101 committed:
Copy Missions folder during Install step

CaptainSwag101 committed:
Fix oapiWriteLogError macro usage in Hsysparse

CaptainSwag101 committed:
Fix oapiWriteLogError macro usage in Hsysparse

CaptainSwag101 committed:
Fix oapiWriteLogError macro usage in Hsysparse

CaptainSwag101 committed:
Add Mission source to MCC project

CaptainSwag101 committed:
Add Mission source to MCC project

CaptainSwag101 committed:
Add Mission source to MCC project

CaptainSwag101 committed:
Enable new lunar gravity model & RTCC support for it

CaptainSwag101 committed:
Enable new lunar gravity model & RTCC support for it

CaptainSwag101 committed:
Enable new lunar gravity model & RTCC support for it

CaptainSwag101 committed:
Use R2 padload values for the AGC where applicable

CaptainSwag101 committed:
Use R2 padload values for the AGC where applicable

CaptainSwag101 committed:
Use R2 padload values for the AGC where applicable

CaptainSwag101 committed:
Enable new Earth gravity model & RTCC support for it

CaptainSwag101 committed:
Enable new Earth gravity model & RTCC support for it

CaptainSwag101 committed:
Enable new Earth gravity model & RTCC support for it

CaptainSwag101 committed:
Disable 32-bit-only moons

While Orbiter 2024 only fully supports 32-bit compilation, 64-bit is an alternate option that is offered and some people may wish to use it. To accommodate this, we will disable the celestial bodies which only exist as closed-source, 32-bit libraries, because they cause havoc with orbits and vessel trajectories when using the 64-bit version of Orbiter 2024. Disabling these only affects NASSP flights, and it will not impact optical sightings because none ever use the moons of the various other planets (nor do the optics have the capability to see them, in all honesty).

CaptainSwag101 committed:
Disable 32-bit-only moons

While Orbiter 2024 only fully supports 32-bit compilation, 64-bit is an alternate option that is offered and some people may wish to use it. To accommodate this, we will disable the celestial bodies which only exist as closed-source, 32-bit libraries, because they cause havoc with orbits and vessel trajectories when using the 64-bit version of Orbiter 2024. Disabling these only affects NASSP flights, and it will not impact optical sightings because none ever use the moons of the various other planets (nor do the optics have the capability to see them, in all honesty).

CaptainSwag101 committed:
Disable 32-bit-only moons

While Orbiter 2024 only fully supports 32-bit compilation, 64-bit is an alternate option that is offered and some people may wish to use it. To accommodate this, we will disable the celestial bodies which only exist as closed-source, 32-bit libraries, because they cause havoc with orbits and vessel trajectories when using the 64-bit version of Orbiter 2024. Disabling these only affects NASSP flights, and it will not impact optical sightings because none ever use the moons of the various other planets (nor do the optics have the capability to see them, in all honesty).

CaptainSwag101 committed:
Update Moon config to latest visual settings / comments

CaptainSwag101 committed:
Update Moon config to latest visual settings / comments

CaptainSwag101 committed:
Update Moon config to latest visual settings / comments

CaptainSwag101 committed:
Update Earth config to latest parameters

Turns out the Earth.cfg file isn't used in modern scenarios, it's the Earth_VirtualAGC file we need to modify!

CaptainSwag101 committed:
Update Earth config to latest parameters

Turns out the Earth.cfg file isn't used in modern scenarios, it's the Earth_VirtualAGC file we need to modify!

CaptainSwag101 committed:
Update Earth config to latest parameters

Turns out the Earth.cfg file isn't used in modern scenarios, it's the Earth_VirtualAGC file we need to modify!

CaptainSwag101 committed:
Remove legacy Earth.cfg and Sol.cfg files

These were only used by the "broken" NASSP 7 scenario holdovers, and they weren't being updated like the rest of our configs. Removing them makes it clear which files need to be changed, and probably improves the situation for the broken scenarios as far as solar system accuracy goes now that they're not using 10-year-old configs.

CaptainSwag101 committed:
Remove legacy Earth.cfg and Sol.cfg files

These were only used by the "broken" NASSP 7 scenario holdovers, and they weren't being updated like the rest of our configs. Removing them makes it clear which files need to be changed, and probably improves the situation for the broken scenarios as far as solar system accuracy goes now that they're not using 10-year-old configs.

CaptainSwag101 committed:
Remove legacy Earth.cfg and Sol.cfg files

These were only used by the "broken" NASSP 7 scenario holdovers, and they weren't being updated like the rest of our configs. Removing them makes it clear which files need to be changed, and probably improves the situation for the broken scenarios as far as solar system accuracy goes now that they're not using 10-year-old configs.

CaptainSwag101 committed:
Fix building with new MCC code

CaptainSwag101 committed:
Fix building with new MCC code

CaptainSwag101 committed:
Fix building with new MCC code

CaptainSwag101 committed:
Add Flag vessel

CaptainSwag101 committed:
Add Flag vessel

CaptainSwag101 committed:
Add Flag vessel

CaptainSwag101 committed:
Add GH Actions workflow for autobuilding

Fix Orbiter dependency path

Fix CMake source path, generator, and architecture

More fixes and improvements to build workflow

Even more fixes?

Fixes pt 3

Final fix?

Set GH token

Fix and simplify release tagging

Try and fix the per-config subdirectory added to the build output

Add comment for buildbot workaround, fix release commit tagging

Use PowerShell directly for downloading Orbiter release ZIP

See if we can use multithreading for compilation

Remove autobuild from PRs, look at CMake branch now

CaptainSwag101 committed:
Add GH Actions workflow for autobuilding

Fix Orbiter dependency path

Fix CMake source path, generator, and architecture

More fixes and improvements to build workflow

Even more fixes?

Fixes pt 3

Final fix?

Set GH token

Fix and simplify release tagging

Try and fix the per-config subdirectory added to the build output

Add comment for buildbot workaround, fix release commit tagging

Use PowerShell directly for downloading Orbiter release ZIP

See if we can use multithreading for compilation

Remove autobuild from PRs, look at CMake branch now

CaptainSwag101 committed:
Add GH Actions workflow for autobuilding

Fix Orbiter dependency path

Fix CMake source path, generator, and architecture

More fixes and improvements to build workflow

Even more fixes?

Fixes pt 3

Final fix?

Set GH token

Fix and simplify release tagging

Try and fix the per-config subdirectory added to the build output

Add comment for buildbot workaround, fix release commit tagging

Use PowerShell directly for downloading Orbiter release ZIP

See if we can use multithreading for compilation

Remove autobuild from PRs, look at CMake branch now

CaptainSwag101 committed:
Bump version number to 9.0

CaptainSwag101 committed:
Bump version number to 9.0

CaptainSwag101 committed:
Bump version number to 9.0

CaptainSwag101 committed:
Remove Appveyor script

CaptainSwag101 committed:
Remove Appveyor script

CaptainSwag101 committed:
Remove Appveyor script

CaptainSwag101 committed:
Update contributing guideline file for Orbiter 2024

CaptainSwag101 committed:
Update contributing guideline file for Orbiter 2024

CaptainSwag101 committed:
Update contributing guideline file for Orbiter 2024

CaptainSwag101 committed:
Try and cache the Orbiter 2024 x64 zip file for the buildbot

CaptainSwag101 committed:
Try and cache the Orbiter 2024 x64 zip file for the buildbot

CaptainSwag101 committed:
Try and cache the Orbiter 2024 x64 zip file for the buildbot

CaptainSwag101 committed:
Remove COPYING.txt since OpenOrbiter is free software

CaptainSwag101 committed:
Remove COPYING.txt since OpenOrbiter is free software

CaptainSwag101 committed:
Remove COPYING.txt since OpenOrbiter is free software

CaptainSwag101 committed:
Remove "beta" from build names, move "x64" to zip filename

CaptainSwag101 committed:
Remove "beta" from build names, move "x64" to zip filename

CaptainSwag101 committed:
Remove "beta" from build names, move "x64" to zip filename

CaptainSwag101 committed:
Update gitattributes to target the new build script

CaptainSwag101 committed:
Update gitattributes to target the new build script

CaptainSwag101 committed:
Update gitattributes to target the new build script

CaptainSwag101 committed:
Update VS version recommendation in readme

CaptainSwag101 committed:
Update VS version recommendation in readme

CaptainSwag101 committed:
Update VS version recommendation in readme

CaptainSwag101 committed:
Add Two-Impulse Processor code files to RTCC and MCC projects

CaptainSwag101 committed:
Add Two-Impulse Processor code files to RTCC and MCC projects

CaptainSwag101 committed:
Add Two-Impulse Processor code files to RTCC and MCC projects

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Update gitignore with zlib

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Update launch scenario descriptions

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
OO Mission Saves

CaptainSwag101 committed:
Fix DirectInput on x64

CaptainSwag101 committed:
Fix DirectInput on x64

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
update direct input

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Apollo 9 DECA cb

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
fix 10 insertion desync

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Documentation' of https://github.com/rcflyinghokie/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Add QSG to builder

rcflyinghokie committed:
Remove "I" and add hyperlink update

Commit:
Merge branch 'O16-SpaceEVA' of https://github.com/Niker25/NASSP.git into O24-SpaceEVA

Commit:
add CMakelists for eva

rcflyinghokie committed:
add versioning

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
versioning syntax

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

Commit:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP into O24-SpaceEVA

Commit:
Merge branch 'O16-SpaceEVA' of https://github.com/Niker25/NASSP.git into O24-SpaceEVA

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

Commit:
Merge branch 'O16-SpaceEVA' of https://github.com/Niker25/NASSP.git into O24-SpaceEVA

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Add notes about EVA control and Apollo 8 PTC during time warp

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into Documentation

rcflyinghokie committed:
Add reflection mode note to 2024 install instructions

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Documentation' of https://github.com/rcflyinghokie/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Add keyboard notes

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'O24-SpaceEVA' of https://github.com/Niker25/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Documentation' of https://github.com/rcflyinghokie/NASSP.git into CMake

rcflyinghokie committed:
Add cue card info and keybinds

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Documentation' of https://github.com/rcflyinghokie/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Add CMake project files

rcflyinghokie committed:
Add optics/crosshair comments

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Documentation' of https://github.com/rcflyinghokie/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Add notes on high res drive downloads

rcflyinghokie committed:
Merge branch 'Documentation' of https://github.com/rcflyinghokie/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'CSMTelecom' of https://github.com/rcflyinghokie/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
add orbiter cdn link

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Documentation' of https://github.com/rcflyinghokie/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Update formatting

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Documentation' of https://github.com/rcflyinghokie/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Spacing/formatting

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Documentation' of https://github.com/rcflyinghokie/NASSP.git into CMake

rcflyinghokie committed:
Merge branch 'Orbiter2016' of https://github.com/orbiternassp/NASSP.git into CMake

CaptainSwag101 committed:
Merge branch 'CMake' of https://github.com/rcflyinghokie/NASSP into CMake
